### DISCUSSION TITLE
What is my Deno version for Supabase Edge Functions?

### DISCUSSION THREAD

**Discussion by thorwebdev** [supabase/supabase#8006] [Questions] [answered]
When running my function locally (supabase functions serve ...`), I can console.log `Deno.version` to get the version, however when deployed, `Deno.version` is `undefined`.

**Comment by thorwebdev** [comment, marked as answer]
We use the Deno CLI to serve your function locally and bundle it for deployment, this is where you can see the version of the Deno CLI being used. We then deploy your function on Deno Deploy, which has a fleet of host machines that run JavaScript VMs (V8 isolates) on demand on the Deno Deploy runtime, which functionally matches the most recent Deno CLI version. 

When we roll out a new update, traffic gradually starts moving from the existing machines to a new set of machines that are running the new version of the Deno Deploy runtime. Whenever we get a request for a given deployment, we’ll assign one of the isolates in the fleet to serve that request. That isolate does that by loading the deployments’ code and then executing that code to serve the request. If we already have an isolate in the fleet that is able to serve the request we use that instead of spinning up a new isolate.

So during rollouts requests will be served by one or more isolates, at one or more runtime versions. This works because the runtime is always backwards compatible (it never contains changes that will break existing deployed code).

The reason for `Deno.version` being undefined when deployed is to avoid adding conditional logic based on the Deno version, to make sure Deno Deploy can automatically upgrade your functions to the most recent version as it's being released.

**Reply by riderx** [reply]
@thorwebdev there are some incompatible versions of Deno with our code, disable us to prevent that will not prevent issues. 
Maybe more the reverse since we have no way to protect ourselves, like when you upgraded and stripe was broken for a month i had to deploy back all my function outside Supabase.
until you fix it, where with version availability i could have provided a fix for certain version.
